client/src/features/LobbyPage/components/GameSettings/GameSettings.tsx

```diff
@@ -1,19 +1,111 @@
-import { FunctionComponent } from "react";
+import { FunctionComponent, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { GameSettingsProps } from "./types";
 import { InputText, Switcher, Timer } from "../../../../components";
+import { AnimeTimerMount } from "../../lib";
+import { GameSettingsActions } from "../../../../redux/GameSettingsReducer/actions";
+import { GameSettingsCurrent, useAppSelector } from "../../../../redux/store";
+import { numberToArr } from "../../../../lib";
 import "./gameSettings.scss";
 
-export const GameSettings: FunctionComponent = (): JSX.Element => {
+export const GameSettings: FunctionComponent<GameSettingsProps> = ({
+  onRegister,
+  onWatch,
+  errors,
+}): JSX.Element => {
+  const settings = useAppSelector(GameSettingsCurrent);
+  const dispatch = useDispatch(); // move all logic in context
+  // refactor toggle func for controll switchers
+  const toggleTimer = () => {
+    dispatch({
+      type: GameSettingsActions.updateSetiings,
+      payload: { isTimerNeed: !settings.isTimerNeed },
+    });
+  };
+  const toggleScrum = () => {
+    dispatch({
+      type: GameSettingsActions.updateSetiings,
+      payload: { masterIsPlayer: !settings.masterIsPlayer },
+    });
+  };
+  const toggleShirt = () => {
+    dispatch({
+      type: GameSettingsActions.updateSetiings,
+      payload: { changingCardInRoundEnd: !settings.changingCardInRoundEnd },
+    });
+  };
+  const baseTime = numberToArr(settings.roundTime);
+  const spShortValue = onWatch("Score type (Short)");
+
+  useEffect(() => {
+    dispatch({
+      type: GameSettingsActions.updateSetiings,
+      payload: { scoreTypeShort: spShortValue },
+    });
+  }, [spShortValue]);
+
   return (
     <>
       <div className="game-settings">
-        <Switcher labelText="Scram master as player:" />
-        <Switcher labelText="Changing card in round end:" />
-        <Switcher labelText="Is timer needed:" />
-        <InputText labelText="Score type:" />
-        <InputText labelText="Score type (Short):" />
-        <div className="game-settings__timer">
-          Round time: <Timer />
-        </div>
+        <Switcher
```

Comment:
3 Switcher component can be wrapped in map functions (just create an array with necessary props as values)